app/src/main/java/com/github/se/travelpouch/ui/notifications/NotificationItem.kt

```diff
@@ -175,36 +177,63 @@ fun handleInvitationResponse(
-        val responseType = if (isAccepted) NotificationType.ACCEPTED else NotificationType.DECLINED
-        val responseMessage = if (isAccepted) "ACCEPTED" else "DECLINED"
-
-        val invitationResponse =
-            Notification(
-                notification.notificationUid,
-                profileViewModel.profile.value.fsUid,
-                notification.senderUid,
-                notification.travelUid,
-                NotificationContent.InvitationResponseNotification(
-                    profileViewModel.profile.value.username, travel!!.title, isAccepted),
-                responseType)
-
-        notificationViewModel.sendNotification(invitationResponse)
-        if (isAccepted) {
-          listTravelViewModel.addUserToTravel(
-              profileViewModel.profile.value.email,
-              travel,
-              { updatedContainer ->
-                listTravelViewModel.selectTravel(updatedContainer)
-                Toast.makeText(context, "User added successfully!", Toast.LENGTH_SHORT).show()
-              },
-              { Toast.makeText(context, "Failed to add user", Toast.LENGTH_SHORT).show() })
-        }
-        Toast.makeText(context, responseMessage, Toast.LENGTH_SHORT).show()
-      },
-      onFailure = { Toast.makeText(context, "Failed to get travel", Toast.LENGTH_SHORT).show() })
+  if (notification.sector == NotificationSector.TRAVEL) {
+    listTravelViewModel.getTravelById(
+        notification.travelUid!!,
+        { travel ->
+          val responseType =
+              if (isAccepted) NotificationType.ACCEPTED else NotificationType.DECLINED
+          val responseMessage = if (isAccepted) "ACCEPTED" else "DECLINED"
+
+          val invitationResponse =
+              Notification(
+                  notification.notificationUid,
+                  profileViewModel.profile.value.fsUid,
+                  notification.senderUid,
+                  notification.travelUid,
+                  NotificationContent.InvitationResponseNotification(
+                      profileViewModel.profile.value.username, travel!!.title, isAccepted),
+                  responseType,
+                  sector = notification.sector)
+
+          notificationViewModel.sendNotification(invitationResponse)
+          if (isAccepted) {
+            listTravelViewModel.addUserToTravel(
+                profileViewModel.profile.value.email,
+                travel,
+                { updatedContainer ->
+                  listTravelViewModel.selectTravel(updatedContainer)
+                  Toast.makeText(context, "User added successfully!", Toast.LENGTH_SHORT).show()
+                },
+                { Toast.makeText(context, "Failed to add user", Toast.LENGTH_SHORT).show() })
+          }
+          Toast.makeText(context, responseMessage, Toast.LENGTH_SHORT).show()
+        },
+        onFailure = { Toast.makeText(context, "Failed to get travel", Toast.LENGTH_SHORT).show() })
+  } else {
+    profileViewModel.addFriend(
```

Comment:
I think that you always add the friend relation, even if the user declined the invitation to be friend. You should test if the response is positive or negative before this line.

Reply:
Yes, I changed it